src/xdocs/config_filters.xml

```diff
@@ -990,9 +1005,133 @@ public class InputTest {
           <a href="https://www.w3schools.com/xml/xpath_syntax.asp">W3Schools Xpath Syntax</a>.
         </p>
         <source>
-&lt;suppress-xpath checks=&quot;.*&quot; query=&quot;//METHOD_DEF[@text='legacyMethod']
+&lt;suppress-xpath checks=&quot;.*&quot; query=&quot;//METHOD_DEF[./IDENT[@text='legacyMethod']]
           /descendant-or-self::node()&quot;/&gt;
         </source>
+
+        <p>Some elements can be suppressed by different ways.
+          For example, to suppress violation on variable <code>wordCount</code> in following code:
+        </p>
+        <source>
+public class InputTest {
+    private int wordCount = 11;
+}
+        </source>
+        <p>You need to look at AST of such code by our CLI tool:</p>
+        <source>
+$ java -jar checkstyle-X.XX-all.jar -t InputTest.java
+CLASS_DEF -> CLASS_DEF [1:0]
+|--MODIFIERS -> MODIFIERS [1:0]
+|   `--LITERAL_PUBLIC -> public [1:0]
+|--LITERAL_CLASS -> class [1:7]
+|--IDENT -> InputTest [1:13]
+`--OBJBLOCK -> OBJBLOCK [1:23]
+|--LCURLY -> { [1:23]
+|--VARIABLE_DEF -> VARIABLE_DEF [2:4]
+|   |--MODIFIERS -> MODIFIERS [2:4]
+|   |   `--LITERAL_PRIVATE -> private [2:4]
+|   |--TYPE -> TYPE [2:12]
+|   |   `--LITERAL_INT -> int [2:12]
+|   |--IDENT -> wordCount [2:16]
+|   |--ASSIGN -> = [2:26]
+|   |   `--EXPR -> EXPR [2:28]
+|   |       `--NUM_INT -> 11 [2:28]
+|   `--SEMI -> ; [2:30]
+`--RCURLY -> } [3:0]
+        </source>
+        <p>The easiest way is to suppress by variable name. As you can see <code>VARIABLE_DEF</code>
+        node refers to variable declaration statement and has child node with token type
+        <code>IDENT</code> which is used for storing class, method, variable names. </p>
+        <p>The following example demonstrates how variable can be queried by its name:</p>
+        <source>
+&lt;suppress-xpath checks=&quot;.&quot; query=&quot;//VARIABLE_DEF[
+            ./IDENT[@text='wordCount']]&quot;/&gt;
+        </source>
+        <p>Another way is to suppress by variable value. Again, if you look at the printed AST tree
+        above, you will notice that one of the grandchildren of <code>VARIABLE_DEF</code> node is
+        responsible for storing variable value - <code>NUM_INT</code> with value <b>11</b>.</p>
+        <p>The following example demonstrates how variable can be queried by its value, same
+        approach applies to <code>String, char, float, double, int, long</code> data types:</p>
+        <source>
+&lt;suppress-xpath checks=&quot;.&quot; query=&quot;//VARIABLE_DEF[.//NUM_INT[@value=11]]&quot;/&gt;
+        </source>
+        <p>Next example is about suppressing method with certain annotation by its value.</p>
```

Comment:
" by its value" -> by its name and element value

Reply:
done